### ISSUE TITLE
False positive for export * as ns

### ISSUE THREAD

**Issue by mjbcopland** [pzavolinsky/ts-unused-exports#160]
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#export--as-ns-syntax

Using ES2020 export-as-namespace syntax fails to detect these exports as used (ie re-exported).

###### something.ts
```
export let something: unknown;
```

###### index.ts
```
export * as Something from "./something";
```

###### output
```
1 module with unused exports
path/to/something.ts: something
```

Using pre-ES2020 syntax of individual import/export statements works as expected. The two are equivalent and should return the same unused exports.

###### index.ts
```
import * as Something from "./something";
export { Something };
```

###### output
```
0 modules with unused exports
```

Re-exporting without a namespace also works as expected. This results in different behaviour but handles `export *` correctly.

###### index.ts
```
export * from "./something";
```

###### output
```
0 modules with unused exports
```

**Comment by mrseanryan** [comment]
hi @mjbcopland thanks for raising this.

Am having trouble reproducing it.

With the 1st example, I get output like:

```
/example/simple/index.ts: Something
```

Which makes sense.

But you have `something.ts: something` - which puzzles me ...

The output I get makes sense, since currently we regard any unused export (or even re-export) as 'unused'.

There is a possible enhancement to add 'library support' to be able to ignore some such exports:

https://github.com/pzavolinsky/ts-unused-exports/issues/105

**Comment by mrseanryan** [comment]
@mjbcopland  - if you have the time, it would be great if you could provide a further full example.

For example a branch that adds example like `https://github.com/pzavolinsky/ts-unused-exports/tree/master/example/simple`

**Comment by mrseanryan** [comment]
Pushed a branch `scratch/export-star-as-issue-160`

```
cd example/simple-index

node ../../bin/ts-unused-exports tsconfig.json
```

OUTPUT: (as expected)

```
1 module with unused exports
/home/sean/src/github/other/ts-unused-exports/example/simple-index/index.ts: Something
```

**Comment by mrseanryan** [comment]
Unless it is something about the `tsconfig.json` ?

**Comment by mrseanryan** [comment]
My `tsconfig.json` is:

```
{
  "compilerOptions": {
    "baseUrl": ".",
    "target": "es5",
    "noEmit": true,
    "noImplicitAny": true,
    "strictNullChecks": true,
    "moduleResolution": "node",
    "paths": {
      "utils": ["utils/src"],
      "utils/*": ["utils/src/*"]
    }
  }
}
```

**Comment by mjbcopland** [comment]
@mrseanryan thanks for looking into this, very interestingly I can confirm that your example works as expected. However, I've discovered how to reproduce the issue, and it has only made me more confused.

Move the source files to a subdirectory, i.e. `src`, leaving tsconfig where it is.

```
renamed:    example/simple-index/index.ts -> example/simple-index/src/index.ts
renamed:    example/simple-index/something.ts -> example/simple-index/src/something.ts
```

```
2 modules with unused exports
ts-unused-exports/example/simple-index/src/index.ts: Something
ts-unused-exports/example/simple-index/src/something.ts: something
```

Where it gets even more confusing is that if you rename the subdirectory to have the same name as the file we're re-exporting from, the output changes again, this time completely different to the other two examples! The value `something` is not recognised as being used, but the namespace `Something` is now being ignored, so we have both the original false-positive on the value _and_ a false-negative on the namespace.

Sidenote: this false-negative is new; is it related or does it need its own bug report?

```
renamed:    example/simple-index/index.ts -> example/simple-index/something/index.ts
renamed:    example/simple-index/something.ts -> example/simple-index/something/something.ts
```

```
1 module with unused exports
ts-unused-exports/example/simple-index/something/something.ts: something
```

Using the `import * as` equivalent works as expected in all scenarios, so there's something weird going on with `export *` in subdirectories, but I'm not sure what.

**Comment by mrseanryan** [comment]
Thanks @mjbcopland for the info.

I think as one bug is OK for now. 

Possibly is internal bug where same named files or folders clobber one another?

**Comment by mrseanryan** [comment]
@mjbcopland - I have a PR that improves support for `export * as X`.

It adds tests, which I believe covers the above.

https://github.com/pzavolinsky/ts-unused-exports/pull/173/files

The issue seemed to be the mechanism that was used to record that file B imports types from file A
(not directly related to sub-folders).

If you have time, you could look at the tests I added, see if they match up ?

OR ideally you could also try that branch, see if it works better for you - branch = `fix/improve-export-star-as`

**Comment by mrseanryan** [comment]
fixed in 7.0.0
